engine/src/execution_graph/logic_controllers/CacheMachine.cpp

```diff
@@ -136,6 +136,10 @@ void CacheMachine::put(size_t index, std::unique_ptr<ral::frame::BlazingTable> t
 	this->addToCache(std::move(table), this->cache_machine_name + "_" + std::to_string(index), true);
 }
 
+void CacheMachine::put(size_t index, std::unique_ptr<ral::cache::CacheData> cacheData) {
+	this->addCacheData(std::move(cacheData), this->cache_machine_name + "_" + std::to_string(index), true);
```

Comment:
this doesnt feel quite right. Is there a reason we keep using these strings concatted to gether for naming  these portions.

Reply:
i am glad you think it does not feel quite right, its gross! but its how we currently can have an array access implementation on top of a queue. I have been wanting to create a new underlying data structure to hold the batches that more natively and permanently allow for both queue APIs and array APIs. OR having two different CacheMachines types that have different APIs. 
I figured we can look into making this better whenever we make more wholesale changes to CacheMachine and/or BlazingTable and/or CacheDAta to allow for multiple kernels to pull the same data from a CacheMachine